### DISCUSSION TITLE
Error: Container #swup not found on page

### DISCUSSION THREAD

**Discussion by daphnevisuals** [swup/swup#646] [Help]
**Description of the issue**

Hi ! 

here is my html (i am using posthtml, so i have an index.html page and a base-layout.html. Here is my index.html, where i want the background-image (in a div) to fade.

```HTML
<div id="swup" class="background-image transition-fade"></div>
  <div class="content">
    <main>
      <!-- contenu du block -->
      <block name="content-homepage">
        <div class="menu">
          <a href="">click me !</a>
          <p>CONTENU</p>
        </div>
      </block>
    </main>
  </div>
```

here is my base-layout page with my nav : 

```HTML
 <body>
    <!-- spécifier le contenu du header ici car il se répète -->
    <header>
      <block name="header">
        <nav>
          <ul>
            <li><a id="mylink" href="./index.html">Accueil</a></li>
            <li><a href="./about.html">À propos</a></li>
            <li><a href="./photography-page.html">Photographie</a></li>
            <li><a href="./webdesign-page.html">Webdesign</a></li>
          </ul>
        </nav>
      </block>
    </header>
    <!-- spécifier tous nos blocks différents qui correspondent aux pages -->
    <block name="content-home"></block>
    <block name="content-about"></block>
    <block name="content-photography"></block>
    <block name="content-webdesign"></block>
    <!-- spécifier le contenu du footer ici car il se répète -->
    <footer>
      <block name="footer">© Daphné Visuals</block>
    </footer>
    <script src="js/main.js" type="module"></script>
    <script src="https://unpkg.com/swup@3"></script>
  </body>
  ```

here is my css : 

```css
.transition-fade {
  transition: 1s;
  opacity: 1;
}

html.is-animating .transition-fade {
  opacity: 0;
}

html.leaving .transition-fade {
  opacity: 0;
}

html.rendering .transition-fade {
  opacity: 1;
}
```

my js : 
```javascript
import Swup from "swup";

const swup = new Swup({
  containers: ["#swup"],
});
```

Console shows this error when i navigate from index to another page.

```
[swup] Container #swup not found on page.
(anonymous) @ about.18dbc454.js:696
g @ about.18dbc454.js:695
enable @ about.18dbc454.js:971
O @ about.18dbc454.js:968
1SICI.@parcel/transformer-js/src/esmodule-helpers.js @ about.18dbc454.js:538
newRequire @ about.18dbc454.js:71
(anonymous) @ about.18dbc454.js:122
(anonymous) @ about.18dbc454.js:145
```

The animation won't apply on the background image and I tried to search why but I don't really understand.
Thanks in advance ! :)

**Comment by daun** [comment]
To make things easier for people to help, I'd suggest creating a test case where people can see the problem or site in action. Currently, it's not very clear what part does what. People unfamiliar with posthtml don't know what `<block>` is, for example. Feel free to use and fork this CodeSandbox template: [codesandbox.io/s/swup-test-case-template-6ci5si](https://codesandbox.io/s/swup-test-case-template-6ci5si)

**Reply by daphnevisuals** [reply]
So i tried to reproduce (with c/c changing the img aswell in the css) what i did in vscode but the extend of base-layout.html doesn't work on the codesandbox.
https://codesandbox.io/s/portfolio-test-swup-posthml-c7v8zd?file=/index.html

**Reply by daphnevisuals** [reply]
sorry, c/c  means cmd c cmd v. I copied my code from vs to codesandbox :p

**Reply by daun** [reply]
Try a version without posthtml required. The idea isn't to have the same code somewhere else — it's to have a minimal version without external dependencies (except swup) so people can make sense of it without all the context.
